core/store/src/store.rs

```diff
@@ -66,6 +111,41 @@ impl Store {
         self.get(column, key)?.as_deref().map(T::try_from_slice).transpose()
     }
 
+    pub fn caching_get_ser<T: BorshDeserialize + Send + Sync + 'static>(
+        &self,
+        column: DBCol,
+        key: &[u8],
+    ) -> io::Result<Option<Arc<T>>> {
+        let Some(column_cache) = self.cache.work_with(column) else {
+            return self.get_ser::<T>(column, key).map(|v| v.map(Into::into));
+        };
+        let mut cache_lock = column_cache.lock();
+        if let Some(value) = cache_lock.get(key) {
+            match Arc::downcast::<T>(Arc::clone(value)) {
+                Ok(result) => return Ok(Some(result)),
+                Err(_) => {
+                    tracing::debug!(
+                        target: "store",
+                        requested = std::any::type_name::<T>(),
+                        "could not downcast an available cached deserialized value"
+                    );
+                }
+            }
+        }
+        // We drop the lock on LRU cache here as the lookup in rocksdb and deserialization can take
+        // quite some time and we don't really want to block other requests from making progress
+        // during this procedure...
+        drop(cache_lock);
+        let value = match self.get_ser::<T>(column, key) {
+            Ok(Some(value)) => Arc::from(value),
+            Ok(None) => return Ok(None),
+            Err(e) => return Err(e),
+        };
+        cache_lock = column_cache.lock();
+        cache_lock.put(key.into(), Arc::clone(&value) as _);
```

Comment:
if we cache items on get, then they would still round trip the DB once. IMO it would be nicer to cache them on write if not too much more complex.